plugins/linear-genome-view/src/LinearGenomeView/components/OverviewScaleBar.tsx

```diff
@@ -78,69 +126,71 @@ const Polygon = observer(
   }) => {
     const theme = useTheme()
     const classes = useStyles()
-    const {
-      offsetPx,
-      width,
-      bpPerPx,
-      dynamicBlocks: visibleRegions,
-      displayedParentRegions,
-    } = model
+    const { offsetPx, width, bpPerPx, dynamicBlocks: visibleRegions } = model
 
+    const blocks = visibleRegions
+      .getBlocks()
+      .filter(block => block.refName !== undefined)
     overview.setVolatileWidth(width)
     overview.showAllRegions()
-
     const polygonColor = theme.palette.tertiary
       ? theme.palette.tertiary.light
       : theme.palette.primary.light
+
+    let bottomLeft: number | undefined
+    let newTopLeft: number | undefined
+    let points: (number | undefined)[][] = []
+    // iterate through blocks and find points for polygon
```

Comment:
can you elaborate on this comment a bit

Reply:
I am iterating over the blocks so that I find the rightmost (endPx and topRight) and leftmost (startPx and topLeft) points to only draw one polygon. Currently we were iterating over each block and returning a polygon (null or drawn polygon).

Reply:
Can you add this to a comment

Reply:
Sure.